taf/updater/updater_pipeline.py

```diff
@@ -1205,6 +1207,45 @@ def _update_tuf_current_revision():
                     target_filepath,
                     current_commit,
                 )
+
+            # TUF updater does not always check the validity of all metadata files
+            # if timestamp is not updated, the updater will determine that a new version
+            # of the snapshot file does not need to be downloaded and it will not be validated
+            # during the update process, the metadata files that TUF updater downloads is stored
+            # in a separate folder within the temp directory
+            # For each commit, check if the metadata files inside that directory are the same
+            # as the ones in the auth repository's metadata folder at that revision
+            pattern = r"\d+\.[^\.\s]+\.\w+"
+            for metadata_file_name in git_updater.get_current_metadata():
```

Comment:
Since we're reading all metadata files from disk for each commit, it's worth timing the updater to see how long this new validation takes, compared to the implementation we have on master.

Reply:
Ran SMC clone a couoek of times. The first was was slightly slower than on master, the second faster than on master, so I don't think that this impacts the performance. Other operations are much slower

Reply:
Awesome